spec/index.bs

```diff
@@ -1894,6 +1900,111 @@ dictionary GPUVertexAttributeDescriptor {
+
+                    1. Let |binding| be a new {{GPUBindGroupLayoutBinding}}.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/binding}} to |resource|'s "binding"
+                        decoration.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                        {{GPUShaderStage/VERTEX}} | {{GPUShaderStage/FRAGMENT}} |
+                        {{GPUShaderStage/COMPUTE}}.
+
+                        Issue: Have a set of all valid {{GPUShaderStage}} on the device.
+
+                    1. If |resource| is for a sampler binding:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/sampler}}.
+
+                    1. If |resource| is for a buffer binding:
+
+                        1. If |resource| is for a uniform buffer:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/uniform-buffer}}.
+
+                        1. If |resource| is for a storage buffer:
+
+                            1. If |resource| is read-only:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/readonly-storage-buffer}}.
+
+                            1. Else:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/storage-buffer}}.
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                                    {{GPUShaderStage/FRAGMENT}} | {{GPUShaderStage/COMPUTE}}.
+
+                    1. If |resource| is for a texture binding:
+
+                        1. Set |binding|.{{GPUBindGroupLayoutBinding/textureComponentType}} to
+                            |resource|'s component type.
+                        1. Set |binding|.{{GPUBindGroupLayoutBinding/textureDimension}} to
+                            |resource|'s dimension.
+                        1. If |resource| is multisampled:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/multisampled}} to true.
+
+                        1. If |resource| is for a sampled texture:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/sampled-texture}}.
+
+                        1. If |resource| is for a storage texture:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/storage-texture}}.
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                                {{GPUShaderStage/FRAGMENT}} | {{GPUShaderStage/COMPUTE}}.
+
+                    1. Append |binding| to |bindings|
+
+            1. For each |b1| in |bindings|:
```

Comment:
Are we doing this in order to support cases where the shader code would have multiple different bindings on the same set/binding?
Given that this ability to derive the layout is only meant for beginners (or lazy people), I would think that just not expecting the bindings to intersect would be most sensible here (as opposed to supporting intersecting but equal bindings).

Reply:
> Are we doing this in order to support cases where the shader code would have multiple different bindings on the same set/binding?

Yes. Also detecting where there would be conflicts. And I just realized this whole algorithm should be put at pipeline creation to run when no `GPUPipelineLayout` is passed.

I don't necessarily agree that this is just for beginners. It can also be for lazy people that know WebGPU and know they are in a code path that doesn't need to be optimal.